src/stores/pacbioLibraries.js

```diff
@@ -0,0 +1,122 @@
+import { defineStore } from 'pinia'
+import useRootStore from '@/stores'
+import { handleResponse } from '@/api/ResponseHelper.js'
+import { groupIncludedByResource, dataToObjectById } from '@/api/JsonApi.js'
+
+export const usePacbioLibrariesStore = defineStore('pacbioLibraries', {
+  state: () => ({
+    tagSetChoices: [],
+    tagChoices: {},
+    libraries: [],
+  }),
+
+  getters: {
+    // Returns a list of tag options by tagSetId
+    tagChoicesForId: (state) => (tagSetId) => {
+      return state.tagChoices[tagSetId] || []
+    },
+  },
+  actions: {
+    /**Create a new libary */
+    async createLibraryInTraction(library, tagId) {
+      const rootState = useRootStore()
+
+      const body = {
+        data: {
+          type: 'libraries',
+          attributes: {
+            pacbio_request_id: library.sample.id,
+            template_prep_kit_box_barcode: library.template_prep_kit_box_barcode,
+            tag_id: tagId,
+            concentration: library.concentration,
+            volume: library.volume,
+            insert_size: library.insert_size,
+            primary_aliquot_attributes: {
+              template_prep_kit_box_barcode: library.template_prep_kit_box_barcode,
+              volume: library.volume,
+              concentration: library.concentration,
+              insert_size: library.insert_size,
+            },
+          },
+        },
+      }
+      //TODO_LIBRARY_CHANGE:Needs to be changed to library
+      const request = rootState.api.traction.pacbio.libraries
+      const promise = request.create({ data: body, include: 'tube,primary_aliquot' })
+      const { success, data: { included = [] } = {}, errors } = await handleResponse(promise)
+      const { tubes: [tube = {}] = [] } = groupIncludedByResource(included)
+      const { attributes: { barcode = '' } = {} } = tube
+      return { success, barcode, errors }
+    },
+
+    /**Delete all libraries with given library ids */
+    async deleteLibraries(libraryIds) {
+      const rootStore = useRootStore()
+      const promises = rootStore.api.traction.pacbio.libraries.destroy(libraryIds)
+      //TODO_LIBRARY_CHANGE: Update all components (which calls deleteLibraries) to use the new response format
+      const responses = await Promise.all(promises.map((promise) => handleResponse(promise)))
+      return responses
+    },
+
+    /**Set libraries in store with those fetched using given filter and page  */
+    async setLibraries(filter, page) {
+      const rootStore = useRootStore()
+      const pacbioLibraries = rootStore.api.traction.pacbio.libraries
+      const promise = pacbioLibraries.get({
+        page,
+        filter,
+        include: 'request,tag,tube',
+      })
+      const response = await handleResponse(promise)
+
+      const { success, data: { data, included = [], meta = {} } = {}, errors = [] } = response
+      const { tubes, tags, requests } = groupIncludedByResource(included)
+
+      if (success) {
+        /* 
```

Comment:
I'm not entirely sure what is going on here but it might be worth working through it to use the up to date methods. It looks like we are returning all of the libraries and then looping through all of them to add relationships. What we would do now is put all of the relationships in the store and then merge them together when the user wants to view a library. In a sense we are doing a lot of unnecessary manipulation.

Reply:
For most of the stores, this seems to be a common pattern, which I thought would be the way the re-formatting is done on client side for the data fetched from store.
 It seems like the current process involves creating a new library object and populating its attributes/fields from relationships in the library object returned by the service.  For eg, a new barcode field is defined for library object which is retrieved from tube->barcode relationship.

I don't have much insight into json api, but just a thought which is possible in  graphql. 
To make it simpler, would it better for the service to handle these complexities of flattening these relations , ensuring that the client receives a simplified object in the expected format?
I might require further inputs to proceed on this.

Reply:
This is an example of where we are doing it differently using side loading / lazy loading https://github.com/sanger/traction-ui/blob/develop/src/stores/pacbioRunCreate.js#L246. It is far more efficent and is less complex in the code.